pkg/lib/get-timesync-backend.py

```diff
@@ -0,0 +1,39 @@
+import os
+import subprocess
+
+# get-timesync-backend - determine which NTP backend timedatectl will
+#                        enable and whether we support it
+
+roots = [ "/etc", "/run", "/usr/local", "/usr/lib" ]
+
+def gather_files(name, suffix):
+    files = { }
+    for r in roots:
+        d = os.path.join(r, name)
+        if os.path.isdir(d):
+            for f in os.listdir(d):
+                if f.endswith(suffix) and not f in files:
+                    files[f] = os.path.join(d, f)
+    return [ files[f] for f in sorted(files.keys()) ]
```

Comment:
I have a feeling this is easier with using glob().

Reply:
How so?

Reply:
Something like this?
```py
roots =  ["/etc", "/run", "/usr/local", "/usr/lib"]
units = itertools.chain(*map(lambda r: glob(f"{r}/systemd/ntp-units.d/*.list"), roots)
unit = first_unit(units)
```

Reply:
Still current, but will copy to new review round, too noisy otherwise.